pkg/ebpf/bytecode/perf.go

```diff
@@ -0,0 +1,37 @@
+//+build linux_bpf
+
+package bytecode
+
+import (
+	"sync"
+
+	"github.com/DataDog/ebpf/manager"
+)
+
+type PerfHandler struct {
+	ClosedChannel chan []byte
+	LostChannel   chan uint64
+	once          sync.Once
+}
+
+func NewPerfHandler(closedChannelSize int) *PerfHandler {
+	return &PerfHandler{
+		ClosedChannel: make(chan []byte, closedChannelSize),
+		LostChannel:   make(chan uint64, 10),
+	}
+}
+
+func (c *PerfHandler) dataHandler(CPU int, batchData []byte, perfMap *manager.PerfMap, manager *manager.Manager) {
```

Comment:
It looks like these calls are passed indirectly to `ebpf/manager`.  Is the shutdown process coordinated enough so that calls to the handler methods are guaranteed to have stopped before `Stop` is called and the channels are closed?

Reply:
Good question. The shutdown happens here: https://github.com/DataDog/datadog-agent/blob/7bc1ec2a457b8d49fc359e78dfdb417a49ed3285/pkg/ebpf/tracer.go#L404-L405

I can add some further coordination within the `PerfHandler` so it doesn't try to send to a closed channel.